src/server/bg_services/agent_blocks_verifier.js

```diff
@@ -70,28 +70,19 @@ class AgentBlocksVerifier {
             });
     }
 
-    populate_agent_blocks_verifier_blocks(blocks) {
+    /**
+     * 
+     * @param {nb.BlockSchemaDB[]} blocks 
+     */
+    async populate_agent_blocks_verifier_blocks(blocks) {
         if (!blocks || !blocks.length) return;
 
-        return this.populate_nodes_for_blocks(blocks)
-            .then(blocks_with_nodes => {
-                const blocks_with_alive_nodes = blocks_with_nodes.filter(block =>
-                    block.node && block.node.rpc_address && block.node.online);
-                if (!blocks_with_alive_nodes || !blocks_with_alive_nodes.length) return;
-                return this.populate_chunks(blocks_with_alive_nodes, 'chunk', { frags: 1, chunk_config: 1 });
-            })
-            .then(populated_blocks => {
-                if (!populated_blocks || !populated_blocks.length) return;
-                return populated_blocks.map(block => {
-                    const chunk_config = this.get_by_id(block.chunk.chunk_config);
-                    if (!chunk_config) {
-                        dbg.warn('populate_agent_blocks_verifier_blocks: Chunk config not found', block.chunk.chunk_config);
-                    }
-                    block.chunk.chunk_coder_config = chunk_config && chunk_config.chunk_coder_config;
-                    const frag = block.chunk.frags.find(frag_rec => String(frag_rec._id) === String(block.frag));
-                    return mapper.get_block_md(block.chunk, frag, block);
-                });
-            });
+        const populated_blocks = await this.populate_nodes_for_blocks(blocks);
+        if (!populated_blocks || !populated_blocks.length) return;
+        const blocks_with_alive_nodes = populated_blocks.filter(block =>
```

Comment:
We populate chunks that we do not need (previously it was divided into two statements).

Reply:
true. but needed in the new architecture, when using BlockApi and BlockDB